switch-base-branch.sh

```
@@ -0,0 +1,293 @@
+    else
+        REMOTE="origin"
+    fi
+}
+
+detect_current_and_target_base() {
+    detect_remotes
+    git fetch $REMOTE --quiet
+    log "[1/4] ${GREEN}✓${NC} Analyzed branches"
+
+    # Find merge-base with both main and release
+    local main_merge_base=$(git merge-base $CURRENT_BRANCH $REMOTE/$MAIN_BRANCH)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)